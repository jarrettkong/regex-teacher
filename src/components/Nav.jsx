import React, { Component } from 'react'
import './Nav.scss';

class Nav extends Component {
  render() {
    return (
      <aside className="Nav">
        <div className="Nav-container">
          <a href="#App" className="home-icon Nav-icon"><i className="material-icons">home</i></a>
          <p>Home</p>
        </div>
        <div className="Nav-container">
          <button className="reset-icon Nav-icon" onClick={this.props.resetProgress}><i className="material-icons">loop</i></button>
          <p>Reset</p>
        </div>
        <div className="Nav-container">
          <a href="https://i.imgur.com/Wn8eOKG.png" target="_blank" rel="noopener noreferrer" className="help-icon Nav-icon"><i className="material-icons">help_outline</i></a>      
          <p>Help</p>
        </div>
      </aside>
    )
  }
}

export default Nav;