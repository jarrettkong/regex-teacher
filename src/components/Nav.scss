$accent-color: #00C296;

.Nav {
  position: absolute;
  top: 30px;
  left: 25px;  
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  & p { 
    display: inline;
    margin: 0 0 0 15px;
  }
}

.Nav-icon {
  background-color: $accent-color;
  border: 0;
  color: white;
  padding: 15px;
  cursor: pointer;
  border-radius: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  margin-bottom: 25px;

  & i {
    font-size: 40px;
  }

  &:hover {
    background-color: adjust-color($color: $accent-color, $lightness: 3%)
  }

  &:focus {
    outline: transparent;
  }
}