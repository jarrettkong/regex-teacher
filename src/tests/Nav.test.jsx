import React from 'react';
import Nav from '../components/Nav.jsx';
import { shallow } from 'enzyme';

describe('Nav', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Nav />
    )
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})