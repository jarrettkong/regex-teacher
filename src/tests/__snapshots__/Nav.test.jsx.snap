// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly 1`] = `
<aside
  className="Nav"
>
  <div
    className="Nav-container"
  >
    <a
      className="home-icon Nav-icon"
      href="#App"
    >
      <i
        className="material-icons"
      >
        home
      </i>
    </a>
    <p>
      Home
    </p>
  </div>
  <div
    className="Nav-container"
  >
    <button
      className="reset-icon Nav-icon"
    >
      <i
        className="material-icons"
      >
        loop
      </i>
    </button>
    <p>
      Reset
    </p>
  </div>
  <div
    className="Nav-container"
  >
    <a
      className="help-icon Nav-icon"
      href="https://i.imgur.com/Wn8eOKG.png"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="material-icons"
      >
        help_outline
      </i>
    </a>
    <p>
      Help
    </p>
  </div>
</aside>
`;
